lib/model/model_test.go

```diff
@@ -3948,3 +3946,46 @@ func (c *alwaysChanged) Seen(fs fs.Filesystem, name string) bool {
 func (c *alwaysChanged) Changed() bool {
 	return true
 }
+
+func TestRequestLimit(t *testing.T) {
```

Comment:
I don't understand how this test works. Explain?

Reply:
... Yes true, this is racy :|  
The idea is that the returned object from `Request` must be closed before a subsequent call to `Request` can return. So I check that before closing, the second `Request` is still blocking, and after closing it returns. However as you say, that may equally be satisfied by a sufficiently long execution time of `Request`.

I could introduce a delay after the second `Request`. It's still racy, but maybe good enough? At least even the current version of the test without a delay does fail without the changes in this PR on my system. And the test failure/flakyness modes are false negatives, not false positives, which is a bit less annoying.

Reply:
@calmh What do you think about the changed test?  
I added a delay of 10 times the duration of the first request, which under normal circumstances should be enough to make sure the second request has time enough to terminate (in general the second request for the same file should even be faster).